File: beakersort.py
#!/usr/bin/env python3

import random
import sys

def is_sorted(arr):
    for i in range(0,len(arr)-1):
        if arr[i] > arr[i+1]:
            return False
    return True

def beakersort(arr):
    while not is_sorted(arr):
        a = 0
        b = 0
        while a == b:
            a = random.randint(0,len(arr)-1)
            b = random.randint(0,len(arr)-1)
        arr[a],arr[b] = arr[b],arr[a]
    return arr

if __name__ == "__main__":
    if len(sys.argv) > 1:
        sorted_arr = beakersort(list(map(lambda x: int(x), sys.argv[1:])))
        print(" ".join(map(lambda x: str(x), sorted_arr)))

File: beakersort.js
#!/usr/bin/env node

function isSorted(array) {
    for (let i = 0; i < array.length - 1; i++) {
        if (array[i] > array[i + 1]) {
            return false;
        }
    }
    return true;
}

function randomInt(min, max) {
    return Math.floor(Math.random() * (max - min)) + min;
}

function beakerSort(input) {
    const array = input.slice();
    
    while (!isSorted(array)) {
        let a = 0;
        let b = 0;
        
        while (a == b) {
            a = randomInt(0, array.length);
            b = randomInt(0, array.length);
        }
        const tmp = array[a];
        
        array[a] = array[b];
        array[b] = tmp;
    }
    return array;
}

// Main
if (process.argv.length > 2) {
    console.log(beakerSort(process.argv.slice(2).map(function (x) { return parseInt(x) })).join(' '));
}
